src/game.rs

```diff
@@ -22,22 +22,66 @@ impl Game {
     /// - An instance of Game with size `dimensions` and all cells in `living_cells` alive.
     /// - An error if `cells` contains out of bound positions with respect to `dimensions`.
     pub fn from_cells(dimensions: Position, cells: &Vec<Position>) -> Result<Self, GameError> {
-        let (width, height) = dimensions;
-        let mut grid = vec![vec![Cell::new(); width]; height];
+        let (rows, columns) = dimensions;
+        let mut grid = vec![vec![Cell::new(); columns]; rows];
 
-        for (cell_i, cell_j) in cells.iter() {
+        for (cell_row, cell_column) in cells.iter() {
             //Return error if the cell position is out of bounds
-            if !Self::in_bounds(*cell_i, *cell_j, width, height) {
-                return Err(GameError::OutOfBoundsGridAccess((*cell_i, *cell_j)));
+            if !Self::in_bounds(*cell_row, *cell_column, rows, columns) {
+                return Err(GameError::OutOfBoundsGridAccess(
+                    (*cell_row, *cell_column),
+                    dimensions,
+                ));
             }
             //Otherwise, give life to the cell and return OK
-            grid[*cell_i][*cell_j].give_life();
+            grid[*cell_row][*cell_column].give_life();
         }
         let alive_cells = cells.clone();
 
         Ok(Game { grid, alive_cells })
     }
 
+    /// Kills the cell if it's alive, gives life to it if it's dead.
+    pub fn toggle_cell(&mut self, pos: (usize, usize)) -> Result<(), GameError> {
+        if !Self::in_bounds(pos.0, pos.1, self.grid.len(), self.grid[0].len()) {
+            return Err(GameError::OutOfBoundsGridAccess(
+                (pos.0, pos.1),
+                (self.grid.len(), self.grid[0].len()),
+            ));
+        }
+
```

Comment:
This out of bounds error check should be abstracted into a function, it's also done in from_cells function